source/particle/world.cc

```diff
@@ -206,6 +210,88 @@ namespace aspect
+                  {
+                    for (unsigned int i = particles_in_cell; i < min_particles_per_cell; ++i)
+                      {
+                        std::pair<aspect::Particle::types::LevelInd,Particle<dim> > new_particle = generator->generate_particle(cell,0);
+
+                        Vector<double> value(this->introspection().n_components);
+                        std::vector<Tensor<1,dim> > gradient (this->introspection().n_components,Tensor<1,dim>());
+
+                        std::vector<Vector<double> >  solution(1,value);
+                        std::vector<std::vector<Tensor<1,dim> > > gradients(1,gradient);
+
+                        std::vector<Point<dim> >     particle_points(1);
+                        particle_points[0] = this->get_mapping().transform_real_to_unit_cell(cell, new_particle.second.get_location());
+
+                        const Quadrature<dim> quadrature_formula(particle_points);
+                        FEValues<dim> fe_value (this->get_mapping(),
+                                                this->get_fe(),
+                                                quadrature_formula,
+                                                update_values |
+                                                update_gradients);
+
+                        fe_value.reinit (cell);
+                        fe_value.get_function_values (this->get_solution(),
+                                                      solution);
+                        fe_value.get_function_gradients (this->get_solution(),
+                                                         gradients);
+
+                        property_manager->initialize_late_particle(new_particle.second,
+                                                                   particles,
+                                                                   *interpolator,
+                                                                   solution[0],
+                                                                   gradients[0]);
+
+                        particles.insert(new_particle);
+                      }
+                  }
+
+                // Remove particles if necessary
+                if ((max_particles_per_cell > 0) && (particles_in_cell > max_particles_per_cell))
+                  {
+                    unsigned int particle_index = 0;
+                    const std::pair<typename std::multimap<types::LevelInd, Particle<dim> >::iterator, typename std::multimap<types::LevelInd, Particle<dim> >::iterator>
+                    particles_in_cell = particles.equal_range(found_cell);
+
+                    for (typename std::multimap<types::LevelInd, Particle<dim> >::iterator particle = particles_in_cell.first;
+                         particle != particles_in_cell.second; ++particle_index)
+                      {
+                        if (particle_index % GeometryInfo<dim>::max_children_per_cell != 0)
```

Comment:
Ok. Just draw randomly which particles you want to remove in that case. I think you already have a random number generator lying around somewhere, then you just need to create an object with a uniform integer distribution between zero and the then current number of particles.